src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -390,18 +393,45 @@ private async Task ProcessSocketAsync(WebSocket socket, Uri url, bool isReconnec
                 {
                     _application.Output.CancelPendingFlush();
                 }
+
+                // Should not throw
+                await receiving.ConfigureAwait(false);
             }
         }
 
-        if (_useStatefulReconnect && !_gracefulClose)
+        var cleanup = true;
+        try
         {
-            if (!UpdateConnectionPair())
+            if (_useStatefulReconnect && !_gracefulClose)
             {
-                return;
-            }
+                if (!UpdateConnectionPair())
+                {
+                    return;
+                }
 
-            await StartAsync(url, _webSocketMessageType == WebSocketMessageType.Binary ? TransferFormat.Binary : TransferFormat.Text,
-                cancellationToken: default).ConfigureAwait(false);
+                try
+                {
+                    await StartAsync(url, _webSocketMessageType == WebSocketMessageType.Binary ? TransferFormat.Binary : TransferFormat.Text,
+                        cancellationToken: default).ConfigureAwait(false);
+                    cleanup = false;
+                }
+                catch
+                {
+                    throw new InvalidOperationException("Reconnect attempt failed.");
```

Comment:
Is this `catch` doing anything valuable other than obscuring the original exception? Why do we want that?

Reply:
I figured if the catch is hit then you tried to connect when you don't have a network connection so it's doomed to fail. It would be nicer to have an error message that the reconnect failed instead of some random transport error?

Although, I could see having the inner exception as the transport error might be nice.